pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/channel/MailboxContentObserver.java

```diff
@@ -84,6 +84,9 @@ public void onNext(MailboxContent mailboxContent) {
           cancelStream();
           break;
         case FIRST_ERROR:
+          _responseObserver.onNext(MailboxStatus.newBuilder().setMailboxId(mailboxId)
+              .putMetadata(ChannelUtils.MAILBOX_METADATA_REQUEST_EARLY_TERMINATE, "true").build());
+          cancelStream();
```

Comment:
Just for my understanding, we're calling `onCompleted` inside the `cancelStream` method to close the gRPC stream (and not channel or connection 🙂)? So then would we never see the `ERROR` case if `FIRST_ERROR` is itself closing the stream?